src/vibrato.ts

```diff
@@ -61,29 +58,34 @@ export class Vibrato extends Modifier {
 
     this.position = Modifier.Position.RIGHT;
     this.renderOptions = {
-      harsh: false,
-      vibratoWidth: 20,
-      waveHeight: 6,
-      waveWidth: 4,
-      waveGirth: 2,
+      code: 0xeab0,
+      width: 20,
     };
 
-    this.setVibratoWidth(this.renderOptions.vibratoWidth);
-  }
-
-  /** Set harsh vibrato. */
-  setHarsh(harsh: boolean): this {
-    this.renderOptions.harsh = harsh;
-    return this;
+    this.setVibratoWidth(this.renderOptions.width);
   }
 
   /** Set vibrato width in pixels. */
   setVibratoWidth(width: number): this {
-    this.renderOptions.vibratoWidth = width;
-    this.setWidth(width);
+    this.renderOptions.width = width;
+    this.text = String.fromCodePoint(this.renderOptions.code);
+    this.measureText();
+
+    const items = Math.round(this.renderOptions.width / this.getWidth());
+    for (let i = 1; i < items; i++) {
+      this.text += String.fromCodePoint(this.renderOptions.code);
+    }
+    this.measureText();
```

Comment:
It took me a while to understand this.... since the constructor() calls this function with this.renderOptions.width and then this function sets this.renderOptions.width to itself... :-)

My question: 

Is the most important part the first `measureText()`, because you have to figure out how wide the character is before you can calculate how many glyphs you need to render?

So something like this wouldn't work?
```ts
const vibratoGlyph = String.fromCodePoint(this.renderOptions.code);
// Render one or more vibrato glyphs.
let t = vibratoGlyph;
const numItems = Math.round(this.renderOptions.width / this.getWidth());
for (let i = 1; i < numItems; i++) {
  t += vibratoGlyph;
}
this.text = t;
this.measureText();
```

Do we have to call measureText() an extra time like this?

```ts
const vibratoGlyph = String.fromCodePoint(this.renderOptions.code);
// Render one or more vibrato glyphs.
let t = vibratoGlyph;

// We need to measureText() so we know the width of a single vibrato glyph.
this.text = t;
this.measureText();

const numItems = Math.round(this.renderOptions.width / this.getWidth());
for (let i = 1; i < numItems; i++) {
  t += vibratoGlyph;
}
this.text = t;
this.measureText();
```

Reply:
The first call to `measureText` calculates the `width` of one glyph. This is used to evaluate how many glyphs we need. The second call evaluates the actual width with `numImtems` Glyphs.